presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/TypeConverter.java

```diff
@@ -199,7 +199,7 @@ private static PrestoToIcebergReturnDataType fromRow(RowType type, int index)
             String name = field.getName().orElseThrow(() ->
                     new PrestoException(NOT_SUPPORTED, "Row type field does not have a name: " + type.getDisplayName()));
             PrestoToIcebergReturnDataType returnType = toIcebergType(field.getType(), index);
-            fields.add(Types.NestedField.required(index, name, returnType.getType()));
+            fields.add(Types.NestedField.optional(index, name, returnType.getType()));
```

Comment:
Can you add a test to `TestIcebergSmoke` that creates a table with a row type, inserts a null into one of the row fields, then reads it?

Reply:
Yes.